drop-storage/src/lib.rs

```diff
@@ -180,7 +180,7 @@ impl Storage {
         remote: Option<sync::FileState>,
         local: Option<sync::FileState>,
     ) -> crate::Result<()> {
-        let mut conn = self.pool.get()?;
+        let mut conn = self.conn.lock().expect("Failed to acquire a lock");
```

Comment:
We could make our own implementation for `self.pool.get()` to return the connection directly instead from the pool thus reducing the PR, this way there would be no need to copy-paste the error. Not a major problem though.

Reply:
I'd say that it would be even better to use `tokio::sync::Mutex` (which does not return `Result`) but decided not to for now since it would break the call sites
@Lipt0nas maybe you can do it in your PR? Since you are already breaking the call sites :P

Reply:
Yeah, ill change it to tokio's mutex in my PR :+1: